src/main/java/RecommendationsServlet.java

```diff
@@ -163,8 +163,9 @@ private void updateFrequencyMaps(ArrayList<VisualizationData> userHistory) {
     // When the user has very similar queries in their search history, the recommendation algorithm
     // may not generate 4. In this case, populate the remaining recommendations of the 5 with random
     // ones.
-    if (recommendationList.size() < 4) {
-      for (int i = 0; i < 5 - recommendationList.size(); i++) {
+    int recommendationSize = recommendationList.size();
+    if (recommendationSize < 5) {
+      for (int i = 0; i < 5 - recommendationSize; i++) {
         recommendationList.add(getRandomRecommendation(userId, userHistory));
```

Comment:
Could this be a while loop (while recommendationList.size() < 5) ?

Reply:
yes it seems more like a while loop.

Reply:
switched to while loop